internal/cli/certificate.go

```diff
@@ -0,0 +1,367 @@
+	base64.StdEncoding.Encode(encoded, csrInner)
+
+	csrResource := &api.CertificateSigningRequest{
+		ApiVersion: "v1alpha1",
+		Kind:       "CertificateSigningRequest",
+		Metadata: api.ObjectMeta{
+			Name: &name,
+		},
+		Spec: api.CertificateSigningRequestSpec{
+			ExpirationSeconds: &expirationSeconds,
+			Request:           csrPEM,
+			SignerName:        o.SignerName,
+			Usages:            &[]string{"clientAuth", "CA:false"},
+		},
+	}
+	csrResourceJSON, err := json.Marshal(csrResource)
+	if err != nil {
+		return nil, err
+	}
+
+	return csrResourceJSON, nil
+}
+
+func submitCsr(name string, c *apiclient.ClientWithResponses, ctx context.Context, csrResourceJSON []byte) error {
+	response, err := c.ReplaceCertificateSigningRequestWithBodyWithResponse(ctx, name, "application/json", bytes.NewReader(csrResourceJSON))
+	if err != nil {
+		return err
+	}
+
+	if response.HTTPResponse != nil {
+		status := response.HTTPResponse.Status
+		statuscode := response.HTTPResponse.StatusCode
+		fmt.Printf("%s: %s\n", status, name)
+		if statuscode != http.StatusOK && statuscode != http.StatusCreated {
+			return fmt.Errorf("%s: failed to apply CertificateSigningRequest: %s", name, string(response.Body))
+		}
+	}
+	return nil
+}
+
+func getCsr(o *CertificateOptions, name string, c *apiclient.ClientWithResponses, ctx context.Context) (*api.CertificateSigningRequest, error) {
+	response, err := c.ReadCertificateSigningRequestWithResponse(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+
+	if response.HTTPResponse != nil {
+		statuscode := response.HTTPResponse.StatusCode
+		if statuscode != http.StatusOK && statuscode != http.StatusCreated {
+			return nil, fmt.Errorf("%s: failed to read CertificateSigningRequest: %s", name, string(response.Body))
+		}
+	}
+
+	currentCsr := api.CertificateSigningRequest{}
+	if err := json.Unmarshal([]byte(string(response.Body)), &currentCsr); err != nil {
+		return nil, err
+	}
+
+	return &currentCsr, nil
+}
+
+func createOutputComponents(name string, currentCsr *api.CertificateSigningRequest, caData []byte) error {
+	cert := currentCsr.Status.Certificate
+	f, err := os.Create(name + ".crt")
+	fmt.Printf("creating output for certificate: %s\n", name+".crt")
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	_, err = f.WriteString(string(*cert))
+	if err != nil {
+		return err
+	}
+
+	f, err = os.Create(name + "-ca-data.crt")
+	fmt.Printf("creating output for CA data: %s\n", name+"-ca-data.crt")
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	_, err = f.WriteString(string(caData))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createOutputConfig(name string, c *apiclient.ClientWithResponses, ctx context.Context, priv crypto.PrivateKey, response *apiclient.EnrollmentConfigResponse) error {
+	filename := "enrollment-config-" + name + ".yaml"
+	f, err := os.Create(filename)
+	fmt.Printf("creating output for enrollment config: %s\n", filename)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
```

Comment:
Can we just print to `stdout`? Then users can redirect to whatever they want to call the file or to pipe to other processors like `yq`? (Obviously, we'd then need to ensure the log output goes to `stderr` instead.)

Reply:
Sure, I'll update